src/index.ts

```diff
@@ -18,6 +15,12 @@ const uuid = (len: number) =>
       return;
     }
 
+    const isBrowser = typeof globalThis.crypto !== 'undefined';
+
+    const randomBytes = isBrowser
+      ? await import('./web-random-bytes').then((mod) => mod.randomBytes)
+      : await import('./node-random-bytes').then((mod) => mod.randomBytes);
```

Comment:
Did you check the JS output from TS?

I don't think this will work without using `node16` or `nodenext` in tsconfig.json

Reply:
the CJS output has 

```
const randomBytes = isWebCryptoFuncDefined
        ? await Promise.resolve().then(() => __importStar(require('./web-random-bytes'))).then((mod) => mod.randomBytes)
        : await Promise.resolve().then(() => __importStar(require('./node-random-bytes'))).then((mod) => mod.randomBytes);
```

I think this is ok, thoughts?

Reply:
That will work for CJS output but it wont work for ESM output

You can't `require()` from ESM

Reply:
sorry, this is the ESM output

```
const randomBytes = isWebCryptoFuncDefined
        ? await import('./web-random-bytes').then((mod) => mod.randomBytes)
        : await import('./node-random-bytes').then((mod) => mod.randomBytes);
```

We have two outputs essentially,

* lib/cjs for our node users
* lib/esm for our browser/edge users